src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,49 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int DEFAULT_VALID_AGE = 18;
+    private static final int DEFAULT_VALID_PASSWORD = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        userValidator(user);
+        loginValidator(user);
+        ageValidator(user);
+        passwordValidator(user);
+        return storageDao.add(user);
+    }
+
+    private void userValidator(User user) {
+        if (user == null) {
+            throw new NullPointerException("Can't register null user!");
+        }
+    }
+
+    private void passwordValidator(User user) {
+        if (user.getPassword() == null) {
+            throw new NullPointerException("User's password cannot be empty");
+        }
+        if (user.getPassword().length() < DEFAULT_VALID_PASSWORD) {
+            throw new RuntimeException("User's password cannot be less then 6 symbols");
```

Comment:
Use constant in messages as well! When client decide to change min password length you just change your constant. And not have to fix messages.